### PR TITLE
machine/stm32f4: refactor common code and add new build tag stm32f4

### PR DESCRIPTION
**Pull request by ardnew**
As requested to make smaller PRs, this is a subset of changes originally proposed in PR #1322.

Specifically, this change creates a new build tag `stm32f4`, adds that tag to the currently one board that supports it (`stm32f4disco`), and refactored code in `stm32f4disco` to a set of common source files usable by `stm32f4disco` and any future boards with this same chip.

### CODE REVIEW

File: src/device/stm32/stm32f4-altfunc-bitfields.go [hunk 1]
```diff
@@ -1,7 +1,6 @@
-// These are the supported alternate function numberings on the stm32f407
-// +build stm32,stm32f407
+// +build stm32f4
```
Comment (sago35) [inline src/device/stm32/stm32f4-altfunc-bitfields.go:4]: Please add the following comments.

https://github.com/tinygo-org/tinygo/blob/26a0819119287713efe95f2ec0c091070c0bd3a3/src/device/stm32/stm32-moder-bitfields.go#L1-L3

File: src/machine/board_stm32.go [hunk 2]
```diff
@@ -1,16 +1,5 @@
-// +build bluepill nucleof103rb stm32f4disco
+// +build bluepill nucleof103rb stm32f4
```
Comment (sago35) [inline src/machine/board_stm32.go:1]: I think this file should be deleted.

**Comment by sago35** [comment]
PR Thank you.
Overall it's very good.
I had no problems with flash on f4disco either.

Please change the review comment mentioned above.

```
$ tinygo flash --target stm32f4disco -size short ./src/examples/blinky1
   code    data     bss |   flash     ram
   3740       4    4276 |    3744    4280
Open On-Chip Debugger 0.10.0+dev-g5c9350d2 (2020-08-07-12:56)
Licensed under GNU GPL v2
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2.cfg is deprecated, please switch to interface/stlink.cfg
Info : auto-selecting first available session transport "hla_swd". To override use 'transport select <transport>'.
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
Info : clock speed 2000 kHz
Info : STLINK V2J25M14 (API v2) VID:PID 0483:374B
Info : Target voltage: 2.906762
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : starting gdb server for stm32f4x.cpu on 3333
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08000748 msp: 0x20001000
Info : Unable to match requested speed 8000 kHz, using 4000 kHz
Info : Unable to match requested speed 8000 kHz, using 4000 kHz
** Programming Started **
Info : device id = 0x10076413
Info : flash size = 1024 kbytes
** Programming Finished **
** Resetting Target **
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
shutdown command invoked
```

**Comment by sago35** [comment]
CI should be fine the next time you push.

File: src/device/stm32/stm32f4-altfunc-bitfields.go (hunk 1 above)
Comment (deadprogram) [reply src/device/stm32/stm32f4-altfunc-bitfields.go:4]: Agreed.

File: src/machine/board_stm32.go (hunk 2 above)
Comment (deadprogram) [reply src/machine/board_stm32.go:1]: Unless there is some functionality that is shared across all boards in this family? Seems like we can leave it in for now.

**Comment by deadprogram** [comment]
All tests are passing, so @ardnew is you can please just make the header change requested by @sago35 this PR seems good to go.

File: src/device/stm32/stm32f4-altfunc-bitfields.go (hunk 1 above)
Comment (deadprogram) [reply src/device/stm32/stm32f4-altfunc-bitfields.go:4]: Thanks @ardnew

**Comment by deadprogram** [comment]
Squashed/merged this PR, thank you very much for this contribution @ardnew

**Pull request merged**